src/format/parsed.rs

```diff
@@ -390,7 +390,7 @@ impl Parsed {
         let (verified, parsed_date) = match (given_year, given_isoyear, self) {
             (Some(year), _, &Parsed { month: Some(month), day: Some(day), .. }) => {
                 // year, month, day
-                let date = NaiveDate::from_ymd_opt(year, month, day).ok_or(OUT_OF_RANGE)?;
+                let date = NaiveDate::from_ymd_opt(year, month, day).map_err(|_| OUT_OF_RANGE)?;
```

Comment:
This could be a proper mapping between the variants of `Error` and `ParsedResult`.
Because this will be one of the first places to touch when converting the parsing code (where it will become just `?`), and because that code is still getting the preparations before we can convert it to `Result`s, I'd like to wait with it until that time.